cmd/collection/main.go

```diff
@@ -373,7 +374,13 @@ func main() {
 		}).
 		// Epoch manager encapsulates and manages epoch-dependent engines as we
 		// transition between epochs
-		Component("epoch manager", func(_ cmd.NodeBuilder, node *cmd.NodeConfig) (module.ReadyDoneAware, error) {
+		Component("epoch manager", func(nodeBuilder cmd.NodeBuilder, node *cmd.NodeConfig) (module.ReadyDoneAware, error) {
+
+			workerPool := workerpool.New(4)
```

Comment:
I am wondering if we should abstract this worker pool by an interface. Atm, the worker pool exposes methods to terminate it. It took a bit of time to figure out how the life-cycle management of the worker pool is implemented, specifically because it looks like any component could just terminate the worker pool ... Though, the current implementation would badly break if some component other than main terminated the pool, but we do not represent this requirement in code.

From a conceptual perspective it would make most sense to me, if the worker pools was part of `VoteCollectors` (and `VoteCollectors` would implement the `ReadyDoneAware` and `Startable` interfaces). 

#### suggestion
* we currently only require two distinct methods for the worker pool:
  * a `Submit` method 
  * and a method to terminate it (`StopWait`). 
* I would suggest we introduce two interfaces (maybe put them in `consensus/hotstuff/vote_collectors.go`)
* `VoteCollectors` has a reference to the worker pool so it can terminate it (for this purpose, `VoteCollectors` would itself need to implement the `ReadyDoneAware` and `Startable` interfaces)
* `VoteAggregator` contains `VoteCollectors`

I have implemented this in PR https://github.com/onflow/flow-go/pull/1702 Though, I must admit that it adds quite some complexity to the code. In hindsight, I am not really sure it is worth it.

Reply:
Thanks for your PR, I think it's well worth it. Originally I was a bit bothered that factory method starts accepting worker pool, not sure if there is a way around it. But let's give it a try, either way it makes start/stop simpler in main code base and tests.